staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go

```diff
@@ -223,6 +224,19 @@ func WithTypeAndObjectMeta(s *schema.Structural) *schema.Structural {
 	return result
 }
 
+// MaxCardinality returns the maximum number of times data conforming to the schema could possibly exist in
+// an object serialized to JSON. For cases where a schema is contained under map or array schemas of unbounded
+// size, this can be used as an estimate as the worst case number of times data matching the schema could be repeated.
+// Note that this only assumes a single comma between data elements, so if the schema is contained under only maps,
+// this estimates a higher cardinality that would be possible.
+func MaxCardinality(s *schema.Structural) uint64 {
+	sz := estimateMinSizeJSON(s) + 1 // assume at least one comma between elements
+	if sz == 0 {
+		return math.MaxUint64
```

Comment:
this is confusing... when would estimateMinSizeJSON return -1, and how would we ever have an item count higher than maxRequestSizeBytes?

Reply:
Right, I'm not sure the zero check can ever run, so it's been removed.